### ISSUE TITLE
Feedback providers should export a VT escape sequence for other Terminals

### ISSUE THREAD

**Issue by StevenBucher98** [PowerShell/PowerShell#21495]
### Summary of the new feature / enhancement

In order to have extra UI or visual changes when feedback providers trigger in PowerShell in other terminals, i.e VSCode terminal, we need way to indicate to the terminal in a general way to notify a feedback provider triggered. We should have a VT escape sequence to notify other terminals, or at least have some logic to determine we are in VSCode and output that sequence when feedback providers trigger.

### Proposed technical implementation details (optional)

Output a VT escape sequence with every feedback provider

**Comment by mklement0** [comment]
> have some logic to determine we are in VSCode 

`$env:TERM_PROGRAM -eq 'vscode'` tells you that.

More generally, `$host.UI.SupportsVirtualTerminal` provides this information for any given host('s current configuration).

**Comment by rhubarb-geek-nz** [comment]
> `$env:TERM_PROGRAM -eq 'vscode'` tells you that.
> 
> More generally, `$host.UI.SupportsVirtualTerminal` provides this information for any given host('s current configuration).

Neither mechanism gets through SSH automatically to a remote session, whereas the TERM environment variable does get forwarded.

```
PS> $host.UI.SupportsVirtualTerminal
True
PS> invoke-command -HostName foo -ScriptBLock { $host.UI.SupportsVirtualTerminal }
False
```

**Comment by mklement0** [comment]
Good point about the commands I mentioned only working locally (not also in remoting sessions).

> the TERM environment variable does get forwarded.

That is true if you use the `ssh` utility, but If you enter an interactive session via `Enter-PSSession -HostName foo`, `TERM` is _not_ set.

 * As an aside: If PowerShell is the default shell for the targeted user, something like
 `TERM=xterm-mono ssh jdoe@someserver` (ditto for `TERM=dumb`) results in a broken interactive experience at least on macOS; see also (your own issue): https://github.com/PowerShell/PSReadLine/issues/3918

While `$host.UI.SupportsVirtualTerminal`  is always `$false` in a remote PowerShell session,
`$PSStyle.OutputRendering` still defaults to `Host` in the _absence_ of a `TERM` variable, and PowerShell commands _do_ produce colored output.

It is debatable whether this is the right default and arguably `Enter-PSSession` too should set `TERM`, but, leaving that aside, user code that produces strings for the interactive command-line editing experience could base its decision as to whether to use VT/ANSI escape sequences on `$PSStyle.OutputRendering -in 'Host', 'Ansi'`
